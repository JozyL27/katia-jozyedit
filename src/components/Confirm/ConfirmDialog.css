.back-drop {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: rgba(0, 0, 0, 0.8);
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.ConfirmDialog {
  background: #fff;
  border-radius: .8vmax;
  padding: .7vmax 10vmin;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.ConfirmDialog__prompt {
  padding-top: 1rem;
}

.ConfirmDialog__question {
  padding: 1rem;
}

.ConfirmDialog__controls {
  margin: 1rem;
}

.ConfirmDialog__controls > * {
  margin: 0 1rem;
}

.fade-in {
	opacity: 1;
	animation-name: fadeInOpacity;
	animation-iteration-count: 1;
	animation-timing-function: ease-in;
	animation-duration: .25s;
}

@keyframes fadeInOpacity {
	0% {
		opacity: 0;
	}
	100% {
		opacity: 1;
	}
}