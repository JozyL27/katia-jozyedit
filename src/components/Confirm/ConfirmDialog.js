import React from 'react';
import './ConfirmDialog.css';

function ConfirmDialog({ prompt, giveAnswer }) {
  return (
    <div className='back-drop fade-in'>
      <div className='ConfirmDialog fade-in'>
        {prompt ? <h1 className='ConfirmDialog__prompt'>{prompt}</h1> : null}
        <h2 className='ConfirmDialog__question'>Are you sure?</h2>
        <div className='ConfirmDialog__controls'>
          <button className='blue-button' onClick={() => giveAnswer(true)}>Yes</button>
          <button className='blue-button' onClick={() => giveAnswer(false)}>No</button>
        </div>
      </div>
    </div>
  )
}
export default ConfirmDialog;