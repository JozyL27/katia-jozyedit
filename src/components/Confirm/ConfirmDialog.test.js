import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import ConfirmDialog from './ConfirmDialog'

describe('<ConfirmDialog />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <MemoryRouter>
          <ConfirmDialog />
      </MemoryRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
            <ConfirmDialog />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})