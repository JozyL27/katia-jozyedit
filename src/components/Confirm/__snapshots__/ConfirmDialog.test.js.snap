// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmDialog /> renders the UI as expected 1`] = `
<div
  className="back-drop fade-in"
>
  <div
    className="ConfirmDialog fade-in"
  >
    <h2
      className="ConfirmDialog__question"
    >
      Are you sure?
    </h2>
    <div
      className="ConfirmDialog__controls"
    >
      <button
        className="blue-button"
        onClick={[Function]}
      >
        Yes
      </button>
      <button
        className="blue-button"
        onClick={[Function]}
      >
        No
      </button>
    </div>
  </div>
</div>
`;
